### ISSUE TITLE
Having a 'Forwarding Writer'

### ISSUE THREAD

**Issue by Git5000** [tinylog-org/tinylog#175]
Applications using libraries with many tags need to configure the output writers for all these tags. You will want some libraries to log to one writer and some others to another writer.

One way of doing this is the multi tag writer. This covers a lot of cases. 

However, in my tests for these cases it always lead to the necessity of having a **forwarding** writer, that is, a writer for one tag sends its output to another writer which then handles the real logging. An example being this `writer2 `configuration which sends all _C1_ tags to the actual _TAG1_ (file output) writer:
```
		Configuration.set("writer2", "forward");
		Configuration.set("writer2.level", "WARN");
		Configuration.set("writer2.tag", "C1");
		Configuration.set("writer2.forward", "TAG1");
```
I have not found any native way in tinylog which supports this. If there is one please tell me. If not I think this could be a feature of general interest.

------------------------------------------------------------------------------------------------------------
All this can of course be implemented in a user defined logging provider. But maybe such a forwarding feature is of general use and a `ForwardWriter `should be part of the library.

Implementation of such a feature would _nearly_ be trivial, the writers `write() `method gets the writers for the given tag from the registry and loops them and forwards the log entry like
```
public void write(LogEntry logEntry) throws Exception {
		 		for (Writer writer:writers) {
					writer.write(logEntry);
				}
```

Unfortunately, the _'nearly easy'_ is the problem as the writer needs to to have the same and right log entry values set. And these to get is not so easy.

You could get them in several ways:

1. You require the user to use the _same format string_ for both writers. Then the same log entry values are extracted. Simple but prone to errors for the user.

2. A kind of neater way would be to allow configuration entries to _reference each other_. Then the value would be automatically be set. This can be quite easily be parsed in the writer creation. E.g. using a special marker e.g. "@" as reference tag (in this example we obtain the format and tag from writer (1). So this is simple to implement and better than (1):
```
		Configuration.set("writer2", "forward");
		Configuration.set("writer2.format", "@writer1.format");
		Configuration.set("writer2.level", "INFO");
		Configuration.set("writer2.tag", "C1");
		Configuration.set("writer2.forward", "@writer1.tag");
```

3. You let the writer know about the `TinylogLoggingProvider` parent (as an interface which providers access to `getWriter(tag)`). It can then query the writer using the `getWriter(tag) `and obtain the original log levels from the other writers `getRequiredLogEntryValues().` This is the cleanest and completely transparent method from the API users view point but unfortunately not so easy to implement in the library. It is of course possible,  but requires a few interfaces or a changed writer constructor (of course you need to keep the old one for compatibility, so you have to handle two constructors in the service loader).

4. You call Logger.tag(forward).info|warn|error|... . This handles the LogEntry correctly. But the log message and the logentries might or might not match.

What is your opinion on this idea? Is this a useful general writer which many users need or leave it to a user logging provider?

**Comment by pmwmedia** [comment]
I haven't completely understood the use case for a forward writer. Assigning multiple tags with different severity levels can be solved quite elegant by your feature request in issue #174. Log entries with different format patterns in the same log file could be confusing when reading a log file.

Could you provide a concrete use case? Maybe it will help me to understand the benefit better.

**Comment by Git5000** [comment]
It could be that the multi level solves the same issue. The ideas developed a bit in parallel. I will think about whether there is a use case for this issue left or whether it can be done with the levels. It would be good if we would not need it since it is much more complicated to implement compared to the multi level.

**Comment by Git5000** [comment]
I looked at the use cases and you are right: If we implement the multi level issue #174 all use cases from above are covered. So you can close this issue #175.

**Comment by pmwmedia** [comment]
Closed in favor of #174
